meshtastic/interfaces/ble/receive_service.py

```
@@ -0,0 +1,279 @@
+    ) -> bool:
+        """Return whether receive loop should continue after a disconnect."""
+        logger.debug("Device disconnected: %s", error_message)
+        should_continue = iface._handle_disconnect(
+            f"read_loop: {error_message}", client=previous_client
+        )
+        if not should_continue:
+            iface._set_receive_wanted(False)
+        return should_continue
+
+    @staticmethod
+    def receive_from_radio_impl(iface: "BLEInterface") -> None:
```

Ruff (0.15.5)

[warning] Too many branches (27 > 12)

(PLR0912)